bundles/org.openhab.persistence.rrd4j/src/main/java/org/openhab/persistence/rrd4j/internal/RRD4jPersistenceService.java

```diff
@@ -249,7 +254,12 @@ public void store(Item item) {
     public Iterable<HistoricItem> query(FilterCriteria filter) {
         String itemName = filter.getItemName();
 
-        RrdDb db = getDB(itemName);
+        RrdDb db = null;
+        try {
+            db = getDB(itemName);
+        } catch (Exception e) {
+            logger.warn("Failed to open rrd4j database '{}' ({})", itemName, e.getClass().getName());
```

Comment:
See above.
```suggestion
            logger.warn("Failed to open rrd4j database '{}' ({})", itemName, e.getClass().getName());
            return List.of();
```

Reply:
I agree, print this only once. It will print the same parameter, once at level warn and then again at level debug.